### PR TITLE
Added createExperiment api doc

### PR DESCRIPTION
**Pull request by bharathappali**
Signed-off-by: bharathappali <bharath.appali@gmail.com>

### CODE REVIEW

**Review by chandrams — CHANGES_REQUESTED** [review]

File: design/API.md [hunk 1]
```diff
@@ -276,3 +276,269 @@ Get the status of autotune.
...
+total_duration : How long a trial experiment should run for ?
+warmup_cycles : Number of iterations for waiting for warmup
+warmup_duration : Time to wait for each iteration
+measurement_cycles : Number of iterations for measuring the metrics
+measurement_duration : Time to collect metrics in each iteration
+
+Deployment Settings consists of settings related to deployment like
+
+deployment_policy : What's the type of policy ? (newDeployment / rollingUpdate), target environment (dev / qa / prod)
+deployment_tracking : Which deployments to track (either prod or training or both)
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/API.md:322]: No the trackers section deals with deployments to track. Deployment policy is only to re deploy existing one or to deploy a new one

File: design/API.md [hunk 2]
```diff
@@ -276,3 +276,269 @@ Get the status of autotune.
...
+
+**Request**
+`GET /createExperiment`
+
+`curl -H 'Accept: application/json' -d <INPUT JSON> http://<URL>:<PORT>/createExperiment`
+
+*Description:*
+
+The API endpoint `createExperiment` expects the data object of type `JSON` with a structure as shown
+below. The Input Json holds details of the configuration to try, settings to that particular deployment
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/API.md:291]: Done

File: design/API.md [hunk 3]
```diff
@@ -276,3 +276,269 @@ Get the status of autotune.
...
+Info section consists of the trial information details like `trial_id`, `trial_num`
+
+Settings section consists of 2 sub-sections 
+
+    - trial_settings
+    - deployment_settings
+
+Trial Settings consists of setting related to trial like 
+
+total_duration : How long a trial experiment should run for ?
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/API.md:312]: In markdown viewer i see it in a different line (IntelliJIDEA) lemme know if issue persists

File: design/API.md [hunk 4]
```diff
@@ -276,3 +276,269 @@ Get the status of autotune.
...
+}
+```
+
+##  getTrialStatus
+Get the status of the experiment by passing `runId` to the endpoint.
+
+**Request**
+`GET /getTrialStatus`
+
+`curl -H 'Accept: application/json' -d <INPUT JSON> http://<URL>:<PORT>/getTrialStatus`
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply design/API.md:529]: Will make the changes

[the remark this answers is not in the document]
Comment (bharathappali) [reply design/API.md:529]: currently its a get request only

File: design/API.md [hunk 5]
```diff
@@ -276,3 +276,268 @@ Get the status of autotune.
...
+        ]
+      }
+    }
+  },
+  "deployments": [
+    {
+      "type" : "training",
+      "deployment_name": "petclinic-sample",
+      "namespace" : "default",
+      "state": "",
```
Comment (chandrams) [inline design/API.md:369]: We need to remove state and result fields from input json as it will be part of output json from EM

**Comment by chandrams** [comment]
@bharathappali - Can you please update the API design doc with latest changes for listTrialStatus API

File: design/API.md [hunk 6]
```diff
@@ -327,3 +327,900 @@ Get the status of autotune.
...
+
+*Sample Input JSON:*
+
+```
+{
+    "settings": {
+        "trial_settings": {
+            "total_duration": "6 min",
+            "measurement_cycles": "3",
+            "warmup_duration": "1min",
```
Comment (chandrams) [inline design/API.md:389]: @bharathappali - Format of duration isn't consistent, please correct it.

**Comment by dinogun** [comment]
Closing as this is stale. Re-open if needed.

**Pull request closed without merging**